src/app/info/EditWikiItemView.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import { DbWikiRow } from "@/databaseUtils";
+import { WikiEditModeButton, WikiItemAccordionSurface } from "@/app/info/StyleComponents";
+import { TextField } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import CancelIcon from "@mui/icons-material/Cancel";
+import SaveAltIcon from "@mui/icons-material/SaveAlt";
+import supabase from "@/supabaseClient";
+import { logErrorReturnLogId } from "@/logger/logger";
+
+interface EditViewProps {
+    rowRef: DbWikiRow;
+    setRowRef: (row?: DbWikiRow) => void;
+    setIsInEditModeRef: (isInEditMode: boolean) => void;
+}
+
+const deleteWikiItem = async (
+    setIsInEditModeRef: (isInEditMode: boolean) => void,
+    setRowRef: (row?: DbWikiRow) => void,
+    rowRef?: DbWikiRow
+): Promise<void> => {
+    if (rowRef) {
```

Comment:
rowRef seems like it's always defined when this is called?

Reply:
line 34 of WikiItemEdit sets the rowData to be undefined on deletion